soap/src/java/com/zimbra/soap/type/GlobalExternalStoreConfig.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * ***** BEGIN LICENSE BLOCK *****
+ * Zimbra Collaboration Suite Server
+ * Copyright (C) 2011, 2012, 2013, 2014, 2016 Synacor, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software Foundation,
+ * version 2 of the License.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License along with this program.
+ * If not, see <https://www.gnu.org/licenses/>.
+ * ***** END LICENSE BLOCK *****
+ */
+
+package com.zimbra.soap.type;
+
+import java.util.Collection;
+import java.util.List;
+
+import javax.xml.bind.annotation.XmlAccessType;
+import javax.xml.bind.annotation.XmlAccessorType;
+import javax.xml.bind.annotation.XmlElement;
+
+import com.google.common.collect.Lists;
+import com.zimbra.common.soap.AdminConstants;
+import com.zimbra.common.soap.GlobalExternalStoreConfigConstants;
+
+@XmlAccessorType(XmlAccessType.NONE)
+public final class GlobalExternalStoreConfig {
+
+	/**
+     * @zm-api-field-description Information about globalS3BucketConfigs
+     */
+    @XmlElement(name=GlobalExternalStoreConfigConstants.E_GLOBAL_S3_BUCKET_CONFIGURATIONS, required=true)
+    private final List <GlobalS3BucketConfiguration> globalS3BucketConfigurations = Lists.newArrayList();
+
+    public void setGlobalS3BucketConfigurations(Collection<GlobalS3BucketConfiguration> list) {
+    	globalS3BucketConfigurations.clear();
+        if (list != null) {
```

Comment:
can also check empty here... We have a combined method to check null and empty for list

Reply:
done!